src/test/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPositionTest.java

```diff
@@ -0,0 +1,114 @@
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.internal.TestUtils;
+
+public class BlockCommentPositionTest {
+
+    @Test
+    public void testJavaDocsRecognition() throws Exception {
+        final List<BlockCommentPositionTestMetadata> metadataList = Arrays.asList(
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnClass.java",
+                        BlockCommentPosition::isOnClass, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnMethod.java",
+                        BlockCommentPosition::isOnMethod, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnField.java",
+                        BlockCommentPosition::isOnField, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnEnum.java",
+                        BlockCommentPosition::isOnEnum, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnConstructor.java",
+                        BlockCommentPosition::isOnConstructor, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnInterface.java",
+                        BlockCommentPosition::isOnInterface, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnAnnotation.java",
+                        BlockCommentPosition::isOnAnnotationDef, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnEnumMember.java",
+                        BlockCommentPosition::isOnEnumConstant, 2)
+        );
+
+        for (BlockCommentPositionTestMetadata metadata : metadataList) {
+            final DetailAST ast = TestUtils.parseFile(
+                    new File(getPath(metadata.getFileName()))
+            );
+            final int matches = foundJavaDocsCount(ast, metadata.getAssertion());
+            assertEquals(metadata.getMatchesNum(), matches);
```

Comment:
I think this line should be inside `foundJavaDocsCount`. I don't think the method needs to return anything.